### ISSUE TITLE
Client timezone is not honored?

### ISSUE THREAD

**Issue by btnguyen2k** [sijms/go-ora#475]
Server:
- Oracle Express v21.3.0-xe (running as container)
- Timezone: +11:00
- Table column type: `TIMESTAMP(0) WITH TIME ZONE`

Client:
- Go v1.21.4
- On same machine, timezone +11:00
- Input value of type `time.Time` with value specifically set to `2021-02-28 23:24:25 +0430` (timezone `Asia/Kabul`).
- `github.com/sijms/go-ora/v2 v2.7.24` used as `database/sql` driver.

Row inserted without error. However, when I queried back the row (with SELECT statement), I got value of `2021-02-28 23:24:25 +1100`. I would expect the returned value would be `2021-02-28 23:24:25 +0430`.

Is this a bug or is there any setting that I have missed?

Edit: Queried directly from sqlplus:

```
SQL> SELECT * FROM test_timezone;

ID       DATA_DATETIMEZ
-------- ---------
001      28-FEB-21 11.24.25 PM +11:00
```

**Comment by sijms** [comment]
as you explain 
oracle has 3 type of time object DATE, timestamp, timestamp with timezone
go has only one type for time.Time

| go type | oracle type |
|------ | -------|
|time.Time | DATE|
|go_ora.TimeStamp | TIMESTAMP|
|go_ora.TimeStampTZ | TIMESTAMP WITH TIME ZONE|

so you should pass time as go_ora.TimeStampTZ

**Comment by sijms** [comment]
ok I will test and update the code then inform you

**Comment by sijms** [comment]
fixed in v2.7.25

**Comment by sijms** [comment]
the solution you suggest give 2 issues:
1- during output parameters, data type should defined correctly so we can not use time.Time parameter which will give type `DATE` when using as output parameter for `TIMESTAMP` or `TIMESTAMP with timezone` field it will produce error
2- passing value `time.Time` which contain timezone information as input for `DATE` or `TIMESTAMP`. oracle will shift date value by timezone offset

so custom go types is needed

**Comment by btnguyen2k** [comment]
Hi @sijms, do you have a test case or code example for issue number 1?

Regarding issue number 2: dealing with date/time is not trivial. Devs should be aware of time zones at server side, client side and session time zone (if database/driver supports).
Then developer and/or application has a few choices to deal with date/time/timezone. One common practice is to normalize everything to UTC before writing to database.

Using custom types is an option, too. However, this should only be used if there is no other option as this option is not portable IMHO. It is obvious that using custom types makes the code less portable. Furthermore, if the application is to support multiple database backends via a unified interface such as database/sql. Then going with common practice such as normalizing data/time to UTC, or using only standard types are much better than using custom types from a specific driver.

Maybe an interim solution for now:
- The driver supports custom types as well as standard time.Time
- If the input time.Time DOES have a timezone (any value other than UTC/GMT+0), internally it converts the input time.Time to a custom type that supports timezone.

**Comment by sijms** [comment]
ok I open the issue again
I will try your suggestion and do more testing before release

**Comment by sijms** [comment]
i finish investigation and find bug related to time zone(input) and server timezone version
i fix it in next release now you can use time.Time{} as input for `DATE`, `TIMESTAMP`, `TIMESTAMP WITH TIME ZONE` and `TIMESTAMP WITH LOCAL TIME ZONE`
the test done for time zones UTC, +3, +8  using oracle servers 10, 11, 12, 19 and 21

**Comment by sijms** [comment]
fixed in v2.8.6
